packages/p2p/src/hooks/useP2PExchangeRate.ts

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useFetch } from '@deriv/api';
+import { useExchangeRate } from '@deriv/hooks';
+
+const useP2PExchangeRate = () => {
```

Comment:
As we discussed over the call, I'm still not okay with the implementation of this hook 😬

The `getP2PRate` method accepts a `non-optional` `currency` parameter meaning it's `necessary` for the method to do the calculation and return the rate, But then in the function body we immediately ignore the parameter and return `override_exchange_rate` which is not depending on the `currency`.

`override_exchange_rate` is based on P2P local currency if I understand it correctly, Meaning we can have this local currency inside the hook itself instead of getting it as a parameter and name the hook something that fits its purpose, Like `useP2PLocalCurrencyRate` or something.

And if the currency we pass to `getP2PRate` method is different in different places then I guess we are adding bug to the code 🤔 Because the base of `override_exchange_rate` is always same thing 🤔

Reply:
this is completely make sense and also @niloofar-deriv mentioned that she is planning to implement it in the way that you raised.